src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -179,14 +175,13 @@ public CaseDTO getCaseById(final UUID caseId, CaseQueryRequestDTO requestParamsD
   }
 
   @Override
-  public List<CaseDTO> getCaseByUPRN(
-      UniquePropertyReferenceNumber uprn, CaseQueryRequestDTO requestParamsDTO)
-      throws CTPException {
+  public List<CaseDTO> getCaseBySampleAttribute(
+      String key, String value, CaseQueryRequestDTO requestParamsDTO) throws CTPException {
     if (log.isDebugEnabled()) {
-      log.debug("Fetching latest case details by UPRN", kv("uprn", uprn));
+      log.debug("Fetching latest case details by {}", key, kv("key", key), kv("value", value));
     }
 
-    return callCaseSvcByUPRN(uprn.getValue(), requestParamsDTO.getCaseEvents());
+    return callCaseSvcBySampleAttribute(key, value, requestParamsDTO.getCaseEvents());
```

Comment:
this method `callCaseSvcBySampleAttribute` is a poor name since we get it from the DB. Suggest doing away with the private method and moving the code here.